tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,286 @@
+"""
+Test the feature of container_checker
+"""
+import logging
+
+import pytest
+
+from pkg_resources import parse_version
+from tests.common import config_reload
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.dut_utils import check_container_state
+from tests.common.helpers.dut_utils import clear_failed_flag_and_restart
+from tests.common.helpers.dut_utils import is_hitting_start_limit
+from tests.common.helpers.dut_utils import is_container_running
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+
+@pytest.fixture(autouse=True, scope='module')
+def config_reload_after_tests(duthost):
+    yield
+    config_reload(duthost)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def check_image_version(duthost):
+    """Skips this test if the SONiC image installed on DUT was 201911 or old version.
+
+    Args:
+        duthost: Host DUT.
+
+    Return:
+        None.
+    """
+    if parse_version(duthost.kernel_version) <= parse_version("4.9.0"):
+        pytest.skip("Test was not supported for 201911 and older image version!")
+
+
+@pytest.fixture(autouse=True, scope="module")
+def update_monit_service(duthost):
+    """Update Monit configuration and restart it.
+
+    This function will first reduce the monitoring interval of container checker
+    from 5 minutes to 2 minutes, restart Monit service without delaying. After
+    testing, these two changes will be rolled back.
+
+    Args:
+        duthost: name of Host DUT.
+
+    Return:
+        None.
+    """
+    logger.info("Reduing the monitoring interval of container_checker.")
+    duthost.shell("sudo sed -i '$s/[2-9]\|[1-9][0-9]\+/2/g' /etc/monit/conf.d/sonic-host")
+    duthost.shell("sudo sed -i '/with start delay 300/s/^./#/' /etc/monit/monitrc")
+    logger.info("Restarting the Monit without delaying.")
+    duthost.shell("sudo systemctl restart monit")
+    yield
+    logger.info("Rolling back the Monit configuration of container checker.")
+    duthost.shell("sudo sed -i '$s/2/5/g' /etc/monit/conf.d/sonic-host")
```

Comment:
what about here? what is the original number is not 5?

Reply:
Use another method to add a temporarily configuration line for `container_checker`.